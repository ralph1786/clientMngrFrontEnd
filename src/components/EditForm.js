import React, { Component } from "react";
import { Link } from "react-router-dom";
import { updateChild } from "../actions/childActions";
import { connect } from "react-redux";

class EditForm extends Component {
  state = {
    name: "",
    image: "",
    age: "",
    address: "",
    allergies: "",
    balance: ""
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleCreateSubmit = e => {
    e.preventDefault();
    // console.log(this.state);
    this.props.updateChild(this.state);
  };

  render() {
    console.log(this.props);
    return (
      <div>
        <Link to="/dashboard">
          <button>Dashboard</button>
        </Link>
        <br />
        <br />
        <form onSubmit={this.handleCreateSubmit}>
          <label>Name</label>
          <input
            name="name"
            type="text"
            value={this.state.name}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Image Url</label>
          <input
            name="image"
            type="text"
            value={this.state.image}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Age</label>
          <input
            name="age"
            type="text"
            value={this.state.age}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Address</label>
          <input
            name="address"
            type="text"
            value={this.state.address}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Allergies</label>
          <input
            name="allergies"
            type="text"
            value={this.state.allergies}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Balance</label>
          <input
            name="balance"
            type="text"
            value={this.state.balance}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <input type="submit" value="Edit Child" />
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateChild: childObj => dispatch(updateChild(childObj))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(EditForm);
