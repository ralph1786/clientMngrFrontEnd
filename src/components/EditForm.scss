.edit-form {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: repeat(12, 1fr);
}

// .login h1 {
//   grid-column: 5/9;
//   text-align: center;
// }

.edit-form form {
  -webkit-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  -moz-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  background-color: rgb(32, 221, 158);
  grid-column: 5/9;
  grid-row: 2/6;
  text-align: center;
  font-size: 1.8em;
  margin: 10px;
  padding: 10px;
}

.edit-form form label,
.edit-form form input {
  margin-top: 5px;
  padding: 5px;
  height: 15px;
}

.edit-form form input[type="text"] {
  //   margin-left: 5px;
  width: 50%;
  font-size: 0.8em;
  outline: none;
  float: right;
  margin-right: 20px;
}

.edit-form form label {
  color: rgb(80, 78, 78);
}

.edit-form form input[type="submit"] {
  width: 50%;
  height: 10%;
  font-size: 20px;
  margin-top: 10px;
  margin-bottom: 10px;
  border-radius: 25px;
  outline: none;
}

.edit-form form input[type="submit"]:hover {
  cursor: pointer;
  background-color: rgb(209, 208, 208);
}

#button-edit {
  margin-left: 50px;
  margin-top: 20px;
  padding: 9px;
  border-radius: 25px;
  font-size: 1.04em;
  background-color: rgb(41, 125, 237);
  color: white;
  outline: none;
}

#button-edit:hover {
  cursor: pointer;
  background-color: rgb(83, 152, 241);
}
