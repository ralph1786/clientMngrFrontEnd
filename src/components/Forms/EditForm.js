import React, { Component, Fragment } from "react";

import { withRouter } from "react-router-dom";
import { updateChild } from "../../actions/editActions";
import { toast } from "react-toastify";
import { connect } from "react-redux";
import CancelButton from "../UI/CancelButton";
import LabelInput from "./LabelInput";

import "./EditForm.scss";

export class EditForm extends Component {
  state = {
    id: this.props.editChild.id,
    name: this.props.editChild.name,
    image: this.props.editChild.image,
    age: this.props.editChild.age,
    address: this.props.editChild.address,
    allergies: this.props.editChild.allergies,
    balance: this.props.editChild.balance
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleEditSubmit = e => {
    e.preventDefault();
    this.props.updateChild(this.state);
    toast.warning(`${this.state.name} was edited!`, {
      position: toast.POSITION.BOTTOM_CENTER
    });
    if (this.props.location.pathname === "/edit") {
      this.props.history.push("/dashboard");
    } else {
      this.props.history.push("parent_dashboard");
    }
  };

  render() {
    return (
      <div className="edit-form animated flipInY">
        <form onSubmit={this.handleEditSubmit}>
          <LabelInput
            labelName="Name"
            name="name"
            inputType="text"
            value={this.state.name}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Profile Image"
            name="image"
            inputType="text"
            value={this.state.image}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Age"
            name="age"
            inputType="text"
            value={this.state.age}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Address"
            name="address"
            inputType="text"
            value={this.state.address}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Allergies"
            name="allergies"
            inputType="text"
            value={this.state.allergies}
            changeHandler={this.handleChange}
          />
          {this.props.location.pathname === "/edit_parent" ? (
            " "
          ) : (
            <Fragment>
              <LabelInput
                labelName="Balance"
                name="balance"
                inputType="text"
                value={this.state.balance}
                changeHandler={this.handleChange}
              />
            </Fragment>
          )}
          <button>Edit Child</button>
          <CancelButton pathname={this.props.location.pathname} />
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    editChild: state.editChild.childToEdit
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateChild: childObj => dispatch(updateChild(childObj))
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(EditForm)
);
