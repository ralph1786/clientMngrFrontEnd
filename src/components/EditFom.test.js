import React from "react";
import { EditForm } from "./EditForm";
import { shallow } from "enzyme";
import sinon from "sinon";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<EditForm editChild={{}} location={{}} />);
});

it("creates a snapshot of EditForm", () => {
  expect(wrapper).toMatchSnapshot();
});

it("form submission works", () => {
  const fakeEvent = {
    preventDefault: () => {}
  };
  wrapper.setProps({ updateChild: () => {}, history: [] });
  expect(
    wrapper.find("form").simulate("submit", fakeEvent).calledOnce
  ).toBeUndefined();
});
