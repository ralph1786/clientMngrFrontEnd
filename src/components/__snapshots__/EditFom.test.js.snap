// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a snapshot of EditForm 1`] = `
<Fragment>
  <div
    className="edit-form animated flipInY"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Name
      </label>
      <input
        name="name"
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <label>
        Image Url
      </label>
      <input
        name="image"
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <label>
        Age
      </label>
      <input
        name="age"
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <label>
        Address
      </label>
      <input
        name="address"
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <label>
        Allergies
      </label>
      <input
        name="allergies"
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <label>
        Balance
      </label>
      <input
        name="balance"
        onChange={[Function]}
        type="text"
      />
      <br />
      <br />
      <button>
        Edit Child
      </button>
      <CancelButton />
    </form>
  </div>
</Fragment>
`;
